File: ABC/python/054/B.py
import sys
N,M = map(int,input().split())
A = []
B = []
for I in range(N):
    A += list(map(str,input().split()))
for J in range(M):
    B += list(map(str,input().split()))

if N < 0 or M < 0 or N > 50 or M > 50 or N < M:
    sys.exit()
print(A)
print(B)

if list(set(A) & set(B)) == B:
    print("Yes")
else:
    print("No")

File: ABC/ruby/054/B.rb
N,M = gets().chomp.split(' ').map { |a| a.to_i }
A = []
B = []
exit if !(( 1 <= M && M <= 50) && ( 1 <= N && N <= 50 ) && ( M <= N ))

(1..N).each {
  A << gets().chomp
}
(1..M).each { |n|
  B << gets().chomp
  if (A[n-1].include? B[n-1]) then next else puts 'No';exit end
}
puts 'Yes'